server/methods/core/cart.js

```diff
@@ -343,6 +344,9 @@ Meteor.methods({
         "_id": cart._id,
         "items.variants._id": variantId
       }, {
+        $addToSet: {
+          isDigital: isDigital
```

Comment:
This is for the different cart instances. first it adds the isDigital property if the variant already exists in cart, and the second time if it's a new instance.